dao/src/test/java/com/epam/esm/dao/impl/CertificateDaoImplTest.java

```diff
@@ -0,0 +1,221 @@
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabase;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.web.WebAppConfiguration;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = DBConfig.class)
+@WebAppConfiguration
+@ActiveProfiles("test")
+class CertificateDaoImplTest {
+
+    public static final String DB_REFRESH_SCRIPT = "data.sql";
+
+    private static final long FIRST_ID = 1L;
+    private static final String FIRST_NAME = "first name";
+    private static final String FIRST_DESCRIPTION = "first description";
+    private static final BigDecimal FIRST_PRICE = BigDecimal.valueOf(100);
+    private static final int FIRST_DURATION = 120;
+    private static final int FIRST_MINUTE = 40;
+
+    private static final long SECOND_ID = 2L;
+    private static final String SECOND_NAME = "second name";
+    private static final String SECOND_DESCRIPTION = "second description";
+    private static final BigDecimal SECOND_PRICE = BigDecimal.valueOf(200);
+    private static final int SECOND_DURATION = 240;
+    private static final int SECOND_MINUTE = 50;
+
+    private static final int YEAR = 2021;
+    private static final int MONTH = 1;
+    private static final int DAY = 25;
+    private static final int HOUR = 7;
+    private static final int SECOND = 10;
+
+    private static final String DB_FIELD_NAME = "name";
+    private static final String DB_FIELD_DESCRIPTION = "description";
+    private static final int RECORD_AFFECTED_ONE = 1;
+    private static final int RECORD_AFFECTED_NONE = 0;
+    private static final long THIRD_ID = 3L;
+
+    private static final String SEARCH_CERTIFICATE_QUERY_PART = "WHERE c.name LIKE concat('%', ?, '%') " +
+            "AND c.description LIKE concat('%', ?, '%') order by c.name ASC";
+
+    private CertificateDao certificateDao;
+    private JdbcTemplate jdbcTemplate;
+    private EmbeddedDatabase embeddedDatabase;
+
+    @Autowired
+    private TagMapper tagMapper;
+    @Autowired
+    private CertificateMapper certificateMapper;
+
+    @BeforeEach
+    void setup() {
+        embeddedDatabase = new EmbeddedDatabaseBuilder()
+                .addScript(DB_REFRESH_SCRIPT)
+                .setType(EmbeddedDatabaseType.H2)
+                .build();
+        jdbcTemplate = new JdbcTemplate(embeddedDatabase);
+        certificateDao = new CertificateDaoImpl(jdbcTemplate, certificateMapper);

```

Comment:
I beileve logic from this method could be handled by Spring and test framework

Reply:
thanks for suggestion. fixed